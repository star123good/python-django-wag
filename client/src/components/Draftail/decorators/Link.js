import PropTypes from 'prop-types';
import React from 'react';

import Icon from '../../Icon/Icon';

import TooltipEntity from '../decorators/TooltipEntity';

const getEmailAddress = mailto => mailto.replace('mailto:', '').split('?')[0];
const getDomainName = url => url.replace(/(^\w+:|^)\/\//, '').split('/')[0];

const Link = props => {
  const { entityKey, contentState } = props;
  const data = contentState.getEntity(entityKey).getData();
  let icon;
  let label;
  let tooltipURL;

  if (data.id) {
    icon = 'link';
    tooltipURL = data.url;
    label = data.url;
  } else if (data.url.startsWith('mailto:')) {
    icon = 'mail';
    tooltipURL = getEmailAddress(data.url);
    label = data.url;
  } else {
    icon = 'link';
    tooltipURL = data.url;
    label = getDomainName(data.url);
  }

  return (
    <TooltipEntity
      {...props}
      icon={<Icon name={icon} />}
      label={label}
      url={tooltipURL}
    />
  );
};

Link.propTypes = {
  entityKey: PropTypes.string.isRequired,
  contentState: PropTypes.object.isRequired,
};

export default Link;
