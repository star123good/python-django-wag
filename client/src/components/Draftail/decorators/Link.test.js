import React from 'react';
import { shallow } from 'enzyme';
import { convertFromHTML, ContentState } from 'draft-js';
import Link from './Link';

describe('Link', () => {
  it('exists', () => {
    expect(Link).toBeDefined();
  });

  it('renders', () => {
    const contentBlocks = convertFromHTML('<h1>aaaaaaaaaa</h1>');
    const contentState = ContentState.createFromBlockArray(contentBlocks);
    const contentStateWithEntity = contentState.createEntity('LINK', 'MUTABLE', { url: 'http://example.com/' });
    const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
    expect(shallow((
      <Link
        entityKey={entityKey}
        contentState={contentStateWithEntity}
      >
        <span>Test children</span>
      </Link>
    ))).toMatchSnapshot();
  });

  it('renders email', () => {
    const contentBlocks = convertFromHTML('<h1>aaaaaaaaaa</h1>');
    const contentState = ContentState.createFromBlockArray(contentBlocks);
    const contentStateWithEntity = contentState.createEntity('LINK', 'MUTABLE', { url: 'mailto:test@example.com' });
    const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
    expect(shallow((
      <Link
        entityKey={entityKey}
        contentState={contentStateWithEntity}
      >
        <span>Test children</span>
      </Link>
    ))).toMatchSnapshot();
  });
});
