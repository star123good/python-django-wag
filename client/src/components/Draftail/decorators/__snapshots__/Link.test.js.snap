// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
<span
  className="RichEditor-link"
  data-tooltip="1"
>
  <Icon
    className=""
    name="icon-link"
    title={null}
  />
  <span>
    Test children
  </span>
</span>
`;

exports[`Link renders email 1`] = `
<span
  className="RichEditor-link"
  data-tooltip="2"
>
  <Icon
    className=""
    name="icon-mail"
    title={null}
  />
  <span>
    Test children
  </span>
</span>
`;
